src/Illuminate/Validation/Validator.php

```diff
@@ -385,21 +385,42 @@ protected function validateAttribute($attribute, $rule)
         // If we have made it this far we will make sure the attribute is validatable and if it is
         // we will call the validation method with the attribute. If a method returns false the
         // attribute is invalid and we will add a failure message for this failing attribute.
-        $validatable = $this->isValidatable($rule, $attribute, $value);
+        if (! $this->isValidatable($rule, $attribute, $value)) {
+            return;
+        }
+
+        if (is_a($rule, RuleContract::class, true)) {
+            if (is_string($rule)) {
+                $rule = $this->resolveRuleFromFQCN($rule, $parameters);
+            }
 
-        if ($rule instanceof RuleContract) {
-            return $validatable
-                    ? $this->validateUsingCustomRule($attribute, $value, $rule)
-                    : null;
+            return $this->validateUsingCustomRule($attribute, $value, $rule);
         }
 
         $method = "validate{$rule}";
 
-        if ($validatable && ! $this->$method($attribute, $value, $parameters, $this)) {
+        if (! $this->$method($attribute, $value, $parameters, $this)) {
             $this->addFailure($attribute, $rule, $parameters);
         }
     }
 
+    /**
+     * Given the FQCN of a rule, resolve a concrete implementation of that rule.
+     *
+     * @param string $className
+     * @param array $parameters
+     * @return RuleContract
+     * @throws RuntimeException
+     */
+    protected function resolveRuleFromFQCN($className, $parameters)
```

Comment:
Maybe a bit shorter?
```suggestion
    protected function resolveRule($className, $parameters)
```